app/src/main/java/ch/epfl/sdp/blindly/main_screen/MatchPageFragment.kt

```diff
@@ -173,6 +173,41 @@ class MatchPageFragment : Fragment(), CardStackListener {
     }
 
     /**
+     * This functions calls the Matching Algorithm to get the potential matches and transforms them
+     * into profiles by calling [createProfilesFromUsers].
+     *
+     * TODO: Use it to retrieve potential matches
+     * Since it is not used yet, I prefer to comment both functions for test coverage purposes.
+     *
+     * @return a list of profiles for the matchActivity
+     */
+    /*@RequiresApi(Build.VERSION_CODES.O)
+    private suspend fun getPotentialMatchesProfiles(): List<Profile> {
+        val matchingAlgorithm = MatchingAlgorithm()
+        val potentialUsers = matchingAlgorithm.getPotentialMatchesFromDatabase()
+
+        return if (potentialUsers == null) {
+            listOf()
+        } else {
+            createProfilesFromUsers(potentialUsers)
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun createProfilesFromUsers(users: List<User>?): List<Profile> {
+        if (users == null) {
+            return listOf()
+        }
```

Comment:
Yes it is, since the function takes a `List<User>?`, which can be null. Otherwise, I could use the "?" notation, but I think this null test handles this better.